tests/ui/drop/lint-tail-expr-drop-order-gated.rs

```diff
@@ -1,15 +1,8 @@
-// This test ensures that `tail_expr_drop_order` does not activate in case Edition 2024 is not used
-// or the feature gate `shorter_tail_lifetimes` is disabled.
```

Comment:
given that you're removing the reasoning behind this test, shouldn't the whole test be removed entirely? alternatively keep this comment